documentation/src/test/java/org/hibernate/validator/referenceguide/chapter07/valueextractor/MultimapKeyValueExtractor.java

```diff
@@ -0,0 +1,22 @@
+//tag::include[]
+package org.hibernate.validator.referenceguide.chapter07.valueextractor;
+
+//end::include[]
+
+import javax.validation.valueextraction.ExtractedValue;
+import javax.validation.valueextraction.ValueExtractor;
+
+import com.google.common.collect.Multimap;
+
+//tag::include[]
+public class MultimapKeyValueExtractor
+		implements ValueExtractor<Multimap<@ExtractedValue ?, ?>> {
+
+	@Override
+	public void extractValues(Multimap<?, ?> originalValue, ValueReceiver receiver) {
+		for ( Object key : originalValue.keySet() ) {
+			receiver.keyedValue( "<multimap key>", key, key );
```

Comment:
I find it a bit weird that we pass the key as a "keyed value" with the key as its own key... I would have expected something like `receiver.iterableValue("<multimap key>", key)`. Maybe it's worth some additional explanation?